shell/packages/accounts-identity/accounts-identity-server.js

```diff
@@ -238,3 +202,62 @@ Accounts.linkIdentityToAccount = function (identityId, accountId) {
                       {$push: {"nonloginIdentities": {id: identityId}}});
 
 }
+
+Meteor.publish("accountsOfIdentity", function (identityId) {
+  check(identityId, String);
+  var hasIdentityCursor =
+      Meteor.users.find({$or: [{_id: identityId},
+                               {_id: this.userId, "loginIdentities.id": identityId},
+                               {_id: this.userId, "nonloginIdentities.id": identityId}]});
+  if (hasIdentityCursor.count() == 0) return;
+  var self = this;
+  hasIdentityCursor.observe({removed: function () { self.stop(); }});
+
+  // We maintain a map from identity IDs to live query handles that track profile changes.
+  var loginIdentities = {};
+
+  function addIdentitiesOfAccount(account) {
+    account.loginIdentities.forEach(function(identity) {
+      if (!(identity.id in loginIdentities)) {
+        var user = Meteor.users.findOne({_id: identity.id});
+        if (user) {
+          SandstormDb.fillInProfileDefaults(user);
+          SandstormDb.fillInIntrinsicName(user);
+          var filteredUser = _.pick(user, "_id", "profile");
+          filteredUser.loginAccountId = account._id;
+          filteredUser.sourceIdentityId = identityId;
+          self.added("loginIdentitiesOfLinkedAccounts", user._id, filteredUser);
+        }
+        loginIdentities[identity.id] =
+          Meteor.users.find({_id: identity.id}, {fields: {profile: 1}}).observeChanges({
+            changed: function (id, fields) {
+              self.changed("loginIdentitiesOfLinkedAccounts", id, fields);
+            }
+          });
+      }
+    });
+  }
+  var cursor = Meteor.users.find({$or: [{"loginIdentities.id": identityId},
+                                        {"nonloginIdentities.id": identityId}]});
+  cursor.forEach(addIdentitiesOfAccount);
```

Comment:
Sorry, my previous comment should have said "move this.ready()", not "remove this.ready()". If this.ready() happens after the observe, then you'll get the right behavior: observe() calls "added" callbacks for all matching documents before returning, i.e. it is equivalent to forEach().